src/international/creepOrganizer.ts

```diff
@@ -42,11 +42,16 @@ InternationalManager.prototype.creepOrganizer = function () {
         // Get the creep's role
 
         const { role } = creep
-        if (!role || role.startsWith('shard')) continue
+        if (!role || role.startsWith('shard')) return
 
         // Assign creep a class based on role
 
-        creep = Game.creeps[creepName] = new creepClasses[role](creep.id)
+        const creepClass = creepClasses[role];
+        function isConstructor(obj: any) {
```

Comment:
What is this constructor function?

Reply:
That's to check if creepClass is a type that can be constructed, so that if the creep's name steps on the name of a function that's not a constructor, the code won't error out.  It's then newed up on line 54.